### PR TITLE
Fix typing error in useArea(..)

### PR DESCRIPTION
**Pull request by evancharlton**
There was a typing area in the Area component, and I fixed similar issues that would likely happen in problems/sectors, too.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -687,20 +679,20 @@ export function useProfileTodo(id: number) {
...
-export function useSector(
-  id: number | undefined,
-  options?: Parameters<typeof useData>[1],
-) {
-  return useData<any>(`/sectors?id=${id}`, {
-    ...(options ?? {}),
-    queryKey: [`/sectors`, { id }],
-  });
+export function useSector(id: number | undefined) {
+  return useData<(Success<"getSectors"> & { sectors: any[] }) | undefined>(
```
Comment (evancharlton) [inline src/api.ts:683]: FYI @jossi87 - `Sector.java` is also missing `getSectors()`, so it's dropped from the swagger definition too

Comment (jossi87) [reply src/api.ts:683]: This is added now.

**Comment by jossi87** [comment]
I'll merge it when the conflict is resolved :)

File: src/api.ts (hunk 1 above)
Comment (evancharlton) [reply src/api.ts]: Thanks!

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "eslint-plugin-react-hooks": "^4.6.0",
     "html-webpack-plugin": "^5.5.3",
     "mini-css-extract-plugin": "^2.7.6",
-    "openapi-typescript": "^6.4.0",
+    "openapi-typescript": "5.4.1",
```
Comment (evancharlton) [inline package.json:36]: FYI @jossi87 - until/unless we bump the Swagger dep on the server side, we can't bump this to v6

**Pull request merged**